### ISSUE TITLE
Fix listing of custom block plugins in apperance form

### ISSUE THREAD

**Issue by asmecher** [pkp/pkp-lib#3607]
See https://forum.pkp.sfu.ca/t/ojs-3-1-1-the-newly-added-custom-blocks-cannot-be-repositioned/40839 for details.

**Comment by asmecher** [comment]
OJS 3.1.1-0 and OMP 3.1.1-0 users can fix this by applying [this patch](https://github.com/pkp/customBlockManager/commit/d16d3887857a627e584d8ab5806c49331a1e0485.diff) in the `plugins/generic/customBlockManager` directory.

If you're not comfortable patching, you can download and install [this version of `plugins/generic/customBlockManager/CustomBlockManagerPlugin.inc.php`](https://raw.githubusercontent.com/pkp/customBlockManager/d16d3887857a627e584d8ab5806c49331a1e0485/CustomBlockManagerPlugin.inc.php) into your installation. (Note that it's not generally a good idea to mix code from github into your installation unless you know what you're doing! This should *only* be done for OJS 3.1.1-0 and OMP 3.1.1-0.)

**Comment by isgrim** [comment]
@asmecher 
Hi Alec,
we stumbled upon this issue after migrating an OJS 2.4.8.2 to 3.1.1-stable.

For us the custom blocks still did not appear in the Settings form, I had to add back the `PluginRegistry::loadCategory` hook in the register method of the CustomBlockManagerPlugin.
This was removed in the commit https://github.com/pkp/customBlockManager/commit/ec5838a0c601d75c86f6d72a0ed8ac9f0cd09bba .

But I am not sure if this was the right fix or it is only a workaround, the custom blocks appeared again in the Settings form after this.

It seems the [BlockPluginsListBuilderHandler](https://github.com/pkp/pkp-lib/blob/master/controllers/listbuilder/settings/BlockPluginsListbuilderHandler.inc.php) only loads the block plugins by category and not all the generic Plugins, which would then register the custom blocks.
But I dont know why this is only an issue with the custom blocks and not the other ones.

Hope this helps cheers,
Nils
